src/nvim/edit.c

```diff
@@ -4757,6 +4758,9 @@ ins_compl_next (
 void pum_ext_select_item(int item, bool insert, bool finish)
 {
   if (!pum_visible() || item < -1 || item >= compl_match_arraysize) {
+    if (compl_started && item == -1) {
+      pum_want.finish = finish;
```

Comment:
perhaps this should be `pum_want.finish |= finish;` ? I don't think `finish=false` should cancel an existing request to finish, it is an implementation flag that a `bool` is set here and it is carried out later.

Reply:
Maybe the function should just immediately return if there is an existing request to finish?

Reply:
that's also fine.